iceoryx2/tests/service_event_tests.rs

```diff
@@ -1402,6 +1402,126 @@ mod service_event {
         assert_that!(listener.try_wait_one().unwrap(), is_some);
     }
 
+    #[test]
+    fn listing_all_notifiers_works<S: Service>() {
+        const NUMBER_OF_NOTIFIERS: usize = 18;
+        let service_name = generate_name();
+        let config = generate_isolated_config();
+        let node = NodeBuilder::new().config(&config).create::<S>().unwrap();
+
+        let sut = node
+            .service_builder(&service_name)
+            .event()
+            .max_notifiers(NUMBER_OF_NOTIFIERS)
+            .create()
+            .unwrap();
+
+        let mut notifiers = vec![];
+
+        for _ in 0..NUMBER_OF_NOTIFIERS {
+            notifiers.push(sut.notifier_builder().create().unwrap());
+        }
+
+        let mut notifier_details = vec![];
+        sut.dynamic_config().list_notifiers(|details| {
+            notifier_details.push(details.notifier_id);
+            CallbackProgression::Continue
+        });
+
+        for notifier in notifiers {
+            assert_that!(notifier_details, contains notifier.id());
+        }
```

Comment:
Please also check that the length of `notifier_details` is the same as `notifier`, else it is not guaranteed to have double entries.

This also applies to the listener tests and other endpoints.